### DISCUSSION TITLE
Stable way of converting Euler to Quaternion and back to Euler

### DISCUSSION THREAD

**Discussion by randomanimedude** [jrouwe/JoltPhysics#589] [Q&A] [answered]
At some point I need to disable physics for body and rotate it using Euler angels. But when I try to enable physics again, I need to give body new rotation before enabling it. If I convert Euler to Quaternion using Quat::sEulerAngles and use resulting quat at BodyInterface::SetPositionAndRotation, method Quat::GetEulerAngles returns weird results.
Example:
[-82.6, -180, -180] -> [0.805408061, -0.181474656, -0.536639929, 0.174362570] -> [2.22298765, 0.929203451, -0.929203510]
I can't figure out where the problem is. Is it with Euler to Quat convertion or the other way?

**Comment by jrouwe** [comment, marked as answer]
It looks like you're passing your angles in degrees but sEulerAngles takes angles in radians and not degrees. Use the DegreesToRadians function. GetEulerAngles returns radians so you need to do RadiansToDegrees to get them back.

But even then the results will not be the same as you're running into what is called [gimbal lock](https://www.youtube.com/watch?v=zc8b2Jo7mno). Rotating around X by -90 degrees, then by -180 degrees around Y and then by -180 degrees around Z is the same as rotating by 90 degrees around X. So you'll see [-90, -180, -180] -> [-0.707107, 0, 0, -0.707107] -> [90, 0, 0]. 

(note that I rounded your -82.6 up to -90 because it is easier to visualize when angles are multiples of 90 degrees)
